### PR TITLE
[telepathy-sasl-signon] Use g_message to log CredentialsNeedUpdate

### PR DESCRIPTION
**Pull request by chriadam**
Previously, we used g_debug to log the CredentialsNeedUpdate event,
but g_debug output is suppressed by default.  This commit changes
it to use g_message instead, which is not suppressed, to ensure
that the event shows up in the journal logs.

This commit also suppresses some warnings by switching to newer
API to avoid deprecation warnings, and marking some function
parameters as unused.

### CODE REVIEW

**Comment by chriadam** [comment]
@special @VDVsx @antseppa My glib knowledge approximates zero, can you guys double check I haven't done anything stupid?

**Comment by VDVsx** [comment]
LGTM, but my glib knowledge is also very low and rusty :)

File: telepathy-sasl-signon/empathy-uoa-auth-handler.c [hunk 1]
```diff
@@ -403,7 +410,9 @@ empathy_uoa_auth_handler_supports (EmpathyUoaAuthHandler *self,
     TpAccount *account)
 {
   const gchar *provider;
-  EmpathySaslMechanism mech;
+
+  //EmpathySaslMechanism mech;
```
Comment (monich) [inline telepathy-sasl-signon/empathy-uoa-auth-handler.c]: Delete this line?

File: telepathy-sasl-signon/empathy-uoa-auth-handler.c [hunk 2]
```diff
@@ -403,7 +410,9 @@ empathy_uoa_auth_handler_supports (EmpathyUoaAuthHandler *self,
...
+  (void)self; // suppress unused parameter warning
```
Comment (monich) [inline telepathy-sasl-signon/empathy-uoa-auth-handler.c]: I prefer to use -Wno-unused-parameter to avoid littering the source code with useless statements

**Comment by monich** [comment]
Couple of comments, otherwise looks sane at the first glance. LGTM

**Comment by chriadam** [comment]
I fixed one comment (removed the "mech" line) but left the other (the two suppress warning lines) because I didn't want to change the build flags specified.  Thanks!

**Pull request merged**